syntax = "proto3";

package trainerpb;

option csharp_namespace = "TrainerApi";

import "google/protobuf/timestamp.proto";

service TrainerService {
    rpc GetTrainer (TrainerByIdRequest) returns (TrainerResponse); //Unary call
}

message TrainerByIdRequest {
    string id = 1;
}

message TrainerResponse{
    string id = 1;
    string name = 2;
    int32 age = 3;
    google.protobuf.Timestamp birthdate = 4;
    repeated Medals medals = 5; //repeated significa que regresara un arreglo o un listado
    google.protobuf.Timestamp created_at = 6;
}

message Medals{
    string region = 1;
    MedalType type = 2;
}

enum MedalType{
    UNKNOW = 0; //siempre manejar un default con valor 0
    GOLD = 1;
    SILVER = 2;
    BRONZE = 3;
}